packages/tangle/scheduler.go

```diff
@@ -122,15 +122,34 @@ func (s *Scheduler) Setup() {
 			return
 		}
 
-		if err := s.SubmitAndReady(messageID); err != nil {
+		if err := s.Submit(messageID); err != nil {
 			if !errors.Is(err, schedulerutils.ErrBufferFull) &&
 				!errors.Is(err, schedulerutils.ErrInboxExceeded) &&
 				!errors.Is(err, schedulerutils.ErrInsufficientMana) {
 				s.Events.Error.Trigger(errors.Errorf("failed to submit to scheduler: %w", err))
 			}
 		}
+
+		if s.isReady(messageID) {
+			if err := s.Ready(messageID); err != nil {
+				s.Events.Error.Trigger(errors.Errorf("failed to mark %s as ready: %w", messageID, err))
+			}
+		}
 	}))
 
+	onMessageScheduledOrConfirmed := func(messageID MessageID) {
+		s.tangle.Storage.Approvers(messageID).Consume(func(approver *Approver) {
```

Comment:
Check if the message has been scheduled already to avoid iterating through approvers twice - after scheduling and confirmation.